p2p/service.py

```diff
@@ -22,7 +22,10 @@ class BaseService(ABC, CancellableMixin):
     # Number of seconds cancel() will wait for run() to finish.
     _wait_until_finished_timeout = 5
 
-    def __init__(self, token: CancelToken=None) -> None:
+    # the custom event loop to run in, or None if the default loop should be used
+    loop: asyncio.AbstractEventLoop = None
```

Comment:
@carver @pipermerriam doesn't assigning `None` on the class level introduce the same issue as discussed here? https://github.com/ethereum/py-evm/issues/1012 So, essentially making all instances run in the same `loop` which may not always be what is assumed?

Reply:
Yes, but that is the desired behavior.  If you want a service running in a different event loop you have to be explicit about that.  Generally, we should only have a single event loop per process.